dtfs-central-api/src/services/state-machines/utilisation-report/event-handlers/delete-payment/delete-payment.event-handler.test.ts

```diff
@@ -326,4 +327,136 @@ describe('handleUtilisationReportPaymentDeletedEvent', () => {
+
+    jest.mocked(feeRecordsAndPaymentsMatch).mockResolvedValue(true);
+
+    // Act
+    await handleUtilisationReportDeletePaymentEvent(report, {
+      transactionEntityManager: mockEntityManager,
+      paymentId,
+      requestSource,
+    });
+
+    // Assert
+    expect(mockEventHandler).toHaveBeenCalledTimes(1);
+    expect(mockEventHandler).toHaveBeenCalledWith(
+      expect.objectContaining({
+        payload: expect.objectContaining({
+          hasCorrections: false,
+        }) as FeeRecordPaymentDeletedEvent,
+      }),
+    );
+  });
+
+  it('should set hasCorrections for each linked fee record independently', async () => {
+    // Arrange
+    const requestSource = aDbRequestSource();
+    const paymentId = 123;
+
+    const report = UtilisationReportEntityMockBuilder.forStatus(RECONCILIATION_IN_PROGRESS).build();
+
+    const linkedFeeRecordEntityOne = FeeRecordEntityMockBuilder.forReport(report).withId(456).withCorrections([]).build();
+    const linkedFeeRecordEntityTwo = FeeRecordEntityMockBuilder.forReport(report).withId(789).build();
+    const correctionEntity = FeeRecordCorrectionEntityMockBuilder.forFeeRecordAndIsCompleted(linkedFeeRecordEntityTwo, true).build();
+    linkedFeeRecordEntityTwo.corrections = [correctionEntity];
+
+    const paymentEntity = PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP)
+      .withId(paymentId)
+      .withFeeRecords([linkedFeeRecordEntityOne, linkedFeeRecordEntityTwo])
+      .build();
+    when(mockFindOne).calledWith(PaymentEntity, expect.anything()).mockResolvedValue(paymentEntity);
+
+    const mockEventHandlerOne = aMockEventHandler();
+    const mockEventHandlerTwo = aMockEventHandler();
+    const feeRecordStateMachineOne = aMockFeeRecordStateMachine(mockEventHandlerOne);
+    const feeRecordStateMachineTwo = aMockFeeRecordStateMachine(mockEventHandlerTwo);
+
+    jest
+      .spyOn(FeeRecordStateMachine, 'forFeeRecord')
+      .mockImplementation((feeRecord) => (feeRecord.id === linkedFeeRecordEntityOne.id ? feeRecordStateMachineOne : feeRecordStateMachineTwo));
+
+    jest.mocked(feeRecordsAndPaymentsMatch).mockResolvedValue(false);
+
+    // Act
+    await handleUtilisationReportDeletePaymentEvent(report, {
+      transactionEntityManager: mockEntityManager,
+      paymentId,
+      requestSource,
+    });
+
+    // Assert
+    expect(mockEventHandlerOne).toHaveBeenCalledTimes(1);
+    expect(mockEventHandlerOne).toHaveBeenCalledWith(
+      expect.objectContaining({
+        payload: expect.objectContaining({
+          hasCorrections: false,
+        }) as FeeRecordPaymentDeletedEvent,
+      }),
+    );
+
+    expect(mockEventHandlerTwo).toHaveBeenCalledTimes(1);
+    expect(mockEventHandlerTwo).toHaveBeenCalledWith(
+      expect.objectContaining({
+        payload: expect.objectContaining({
+          hasCorrections: true,
+        }) as FeeRecordPaymentDeletedEvent,
+      }),
+    );
+  });
```

Comment:
could anything go in a beforeEach? these its are pretty long

Reply:
I'll try and refactor the test file a bit